repotracker/repotracker.go

```diff
@@ -693,130 +705,127 @@ func sanityCheckOrderNum(revOrderNum int, projectId, revision string) error {
+				})
+				return errors.Wrapf(err, "error inserting build %s", buildId)
 			}
-			continue
-		}
 
-		var lastActivated *model.Version
-		lastActivated, err = model.VersionFindOne(model.VersionByLastVariantActivation(ref.Identifier, buildvariant.Name))
-		if err != nil {
-			return errors.Wrap(err, "problem getting activatation time for variant")
-		}
+			var lastActivated *model.Version
+			lastActivated, err = model.VersionFindOne(model.VersionByLastVariantActivation(ref.Identifier, buildvariant.Name))
+			if err != nil {
+				return errors.Wrap(err, "problem getting activatation time for variant")
+			}
 
-		var lastActivation *time.Time
-		if lastActivated != nil {
-			for _, buildStatus := range lastActivated.BuildVariants {
-				if buildStatus.BuildVariant == buildvariant.Name && buildStatus.Activated {
-					lastActivation = &buildStatus.ActivateAt
-					break
+			var lastActivation *time.Time
+			if lastActivated != nil {
+				for _, buildStatus := range lastActivated.BuildVariants {
+					if buildStatus.BuildVariant == buildvariant.Name && buildStatus.Activated {
+						lastActivation = &buildStatus.ActivateAt
+						break
+					}
 				}
 			}
-		}
 
-		var activateAt time.Time
-		if lastActivation == nil {
-			// if we don't have a last activation time then prepare to activate it immediately.
-			activateAt = time.Now()
-		} else {
-			activateAt = lastActivation.Add(time.Minute * time.Duration(ref.GetBatchTime(&buildvariant)))
+			var activateAt time.Time
+			if lastActivation == nil {
+				// if we don't have a last activation time then prepare to activate it immediately.
+				activateAt = time.Now()
+			} else {
+				activateAt = lastActivation.Add(time.Minute * time.Duration(ref.GetBatchTime(&buildvariant)))
+			}
+
+			grip.Debug(message.Fields{
+				"message": "activating build",
+				"name":    buildvariant.Name,
+				"project": ref.Identifier,
+				"version": v.Id,
+				"time":    activateAt,
+				"runner":  RunnerName,
+			})
+			v.BuildIds = append(v.BuildIds, buildId)
+			v.BuildVariants = append(v.BuildVariants, model.VersionBuildStatus{
+				BuildVariant: buildvariant.Name,
+				Activated:    false,
+				ActivateAt:   activateAt,
+				BuildId:      buildId,
+			})
 		}
 
-		grip.Debug(message.Fields{
-			"message": "activating build",
-			"name":    buildvariant.Name,
-			"project": ref.Identifier,
+		_, err = evergreen.GetEnvironment().DB().Collection(model.VersionCollection).InsertOne(sessCtx, v)
+		if err != nil {
+			abortErr := sessCtx.AbortTransaction(sessCtx)
+			grip.Notice(message.Fields{
+				"message":    "aborting transaction",
+				"cause":      "can't insert version",
+				"version":    v.Id,
+				"insert_err": err.Error(),
+				"abort_err":  abortErr.Error(),
+			})
+			return errors.Wrapf(err, "error inserting version %s", v.Id)
+		}
+		err = sessCtx.CommitTransaction(sessCtx)
+		if err != nil {
+			return errors.Wrapf(err, "error committing transaction for version %s", v.Id)
```

Comment:
This should probably also log. Since we've not used transactions before, we don't know the fraction of the time that they might abort. It'd be useful to track this, in case it slows down version creation a lot.

Reply:
Added a log statement